src/tensors/vectors.jl

```diff
@@ -0,0 +1,23 @@
+struct OffByOneVector{T} <: AbstractVector{T}
+    data::AbstractVector{T}
+end
+
+function Base.getindex(vector::OffByOneVector{T}, index::Int) where {T}
+    return vector.data[index] + 1
+end
+
+function Base.getindex(vector::OffByOneVector{T}, index::Vararg{Int}) where {T}
+    return vector.data[index...] + 1
```

Comment:
add and subtract 0x01, UInt8 won't widen smaller ints with promotion.

Reply:
Done!

Reply:
Did you mean `... + 1 + 0x01 - 0x01`?

Reply:
a similar approach is `vector.data[index...] + T(1)`